network/retrieve/peer.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package retrieve
+
+import (
+	"github.com/ethersphere/swarm/log"
+	"github.com/ethersphere/swarm/network"
+)
+
+// Peer is the Peer extension for the streaming protocol
+type Peer struct {
+	*network.BzzPeer
+}
+
+// NewPeer is the constructor for Peer
+func NewPeer(peer *network.BzzPeer) *Peer {
+	p := &Peer{
+		BzzPeer: peer,
+	}
+	return p
+}
+func (p *Peer) logError(msg string, ctx ...interface{}) {
```

Comment:
I think it'd be nicer if you have a logger as part of the protocol, rather that define all the necessary methods, such as `logError`, `logDebug`, `logWarn`, `logTrace`, etc. Then you just use the logger from `Retrieval`, which has the additional context.

Reply:
done